### ISSUE TITLE
Feedback on No Application Targets

### ISSUE THREAD

**Issue by tooolbox** [artemnovichkov/Carting#61]
Thanks for the great tool!

I have a (pre-existing) project, a framework, that uses Carthage.  I clone it out, do `carthage update --platform ios` and then `carting update`.

> 🤷‍♂️ Nothing to update.

Hm...there's no `Carthage` script, no `xcfilelist`s or anything.  Building the project errors because the frameworks are missing.  It seems like `carting update` is a complete no-op.
```
carting info
```
I get some results, for example:
> SwiftSoup.framework		dynamic	i386, x86_64, armv7, arm64

Okay, so Carting knows that the frameworks are there, why is `update` not doing anything?

In fooling around, I tried the `--target` flag giving the name of my framework target within the project, and I get `Error: There is no target with <argument> name.`

I feel like I may be doing something wrong or have some rather large misconception; I'm used to `carthage update`, drag the frameworks into Frameworks and Libraries, and `carting update` to finish it off.  I confess to be fairly ignorant as to the deep inner workings of linking frameworks and such business.  Any help would be appreciated and if there's a bug I can do a PR.

-------------------

Totally disrelated, but I spotted it whilst fooling around:

```
~/g/project> carting help update
OVERVIEW: Adds a new script with input/output file paths or updates the script named `Carthage`.

USAGE: carting update [--script <script>] [--path <path>] [--format <format>] [--target <target>] [--project-names <project-names> ...]

OPTIONS:
  -s, --script <script>   The name of Carthage script. (default: Carthage)
  -p, --path <path>       The project directory path. (default: nil)
  -f, --format <format>   Format of input/output file paths: file - using simple paths, list - using xcfilelists (default: list)
  -t, --target <target>   The project directory path. (default: nil)
  -n, --project-names <project-names>
                          The names of projects.
  -h, --help              Show help information.
```

The description for the `--target` flag seems wrong!  It's the same as the `--path` flag.

**Comment by artemnovichkov** [comment]
Hi, @tooolbox! Thanks for using Carting and the feedback!
- About updates - could you please share your Cartfile? I'll try to handle the issue in empty project.
- Target flag is used for name of targets in your project. And it has wrong description, i'll fix it.

**Comment by tooolbox** [comment]
Okay, here's what I'm doing:

1. Clone out https://github.com/readium/r2-navigator-swift
2. `cd r2-navigator-swift`
3. `carthage update --platform ios`
4. The framework already appears in `Targets > r2-navigator-swift > General > Frameworks and Libraries`, but I can drag it there again from where Carthage has built it.
5. Change it from `Embed & Sign` to `Do Not Embed`.
6. Run `carting update` > `🤷‍♂️ Nothing to update.`

Hm...

The same thing occurs if I want to add [SwiftSoup](https://github.com/scinfu/SwiftSoup) to the project.  I make the cartfile look like this:
```
github "readium/r2-shared-swift" == 1.4.3
github "scinfu/SwiftSoup" == 2.3.1
```

I can do the same thing of `carthage update`, drag in the framework, `carting update` and nothing.

Thanks for your help!  Sorry in advance if this is something silly.

**Comment by tooolbox** [comment]
Oh, I see it: https://github.com/artemnovichkov/Carting/blob/master/Sources/CartingCore/Services/ProjectService.swift#L235

You're filtering for *application* targets.  The project I'm using has a *framework* target, i.e. it builds a framework that is used by other projects.  Is it intentional to only pay attention to application targets?

**Comment by tooolbox** [comment]
Just to take my rubber-ducking to a full completion here...

The "copy frameworks" script is only needed when you're creating an application.  When you're creating a framework, you're not going to embed dependencies into your framework.  Instead, when you are creating your final application, which may have a nested tree of framework dependencies, Carthage will [scan the whole tree and bring everything in for you](https://github.com/Carthage/Carthage#nested-dependencies):

> If the framework you want to add to your project has dependencies explicitly listed in a Cartfile, Carthage will automatically retrieve them for you. You will then have to drag them yourself into your project from the Carthage/Build folder.

> If the embedded framework in your project has dependencies to other frameworks you must link them to application target (even if application target does not have dependency to that frameworks and never uses them).

So all you need to do when building a framework is `carthage update` and then drag the frameworks in--the three steps that `carting update` helps with aren't even necessary in this scenario.

See?  Told you I had a large misconception 😉 

At this point I would only ask for better/more feedback from Carting.  Like:
```
$ carting update
Your project has 3 targets, but none of them are Application targets!
🤷‍♂️ Nothing to update.
```

Thanks!

**Comment by artemnovichkov** [comment]
Thanks for the explanation of your case! Of course, we should improve messages and maybe update README with description and examples.

**Comment by artemnovichkov** [comment]
I've released 2.1.3 version. Could you please check the error messages for your case?

**Comment by tooolbox** [comment]
I got: `Error: There are no application targets`

Beautiful!  Thanks very much for this, closing.
